cmd/cloud/deployment_workerqueue_test.go

```diff
@@ -0,0 +1,238 @@
+		cmdArgs := []string{"worker-queue", "create", "-d", "test-deployment-id", "-t", "test-instance-type", "-n", "test-queue"}
+		actualOut, err := execDeploymentCmd(cmdArgs...)
+		assert.NoError(t, err)
+		assert.Equal(t, expectedoutput, actualOut)
+		mockClient.AssertExpectations(t)
+	})
+	t.Run("create worker queue when deployment name was provided", func(t *testing.T) {
+		expectedoutput := "worker queue test-queue for test-deployment-label in ck05r3bor07h40d02y2hw4n4v workspace created\n"
+		// mock os.Stdin
+		expectedInput := []byte("1")
+		r, w, err := os.Pipe()
+		assert.NoError(t, err)
+		_, err = w.Write(expectedInput)
+		assert.NoError(t, err)
+		w.Close()
+		stdin := os.Stdin
+		// Restore stdin right after the test.
+		defer func() { os.Stdin = stdin }()
+		os.Stdin = r
+
+		mockClient.On("ListDeployments", mock.Anything).Return(deploymentRespDefaultQueue, nil).Twice()
+		mockClient.On("GetWorkerQueueOptions").Return(mockWorkerQueueDefaultOptions, nil).Once()
+		mockClient.On("UpdateDeployment", mock.Anything).Return(deploymentRespDefaultQueue[0], nil).Once()
+		cmdArgs := []string{"worker-queue", "create", "--deployment-name", "test-deployment-label", "-t", "test-instance-type", "-n", "test-queue"}
+		actualOut, err := execDeploymentCmd(cmdArgs...)
+		assert.NoError(t, err)
+		assert.Equal(t, expectedoutput, actualOut)
+		mockClient.AssertExpectations(t)
+	})
+	t.Run("create worker queue when no name was provided", func(t *testing.T) {
+		expectedoutput := "worker queue test-queue for test-deployment-label in ck05r3bor07h40d02y2hw4n4v workspace created\n"
+		// mock os.Stdin
+		expectedInput := []byte("test-queue")
+		r, w, err := os.Pipe()
+		assert.NoError(t, err)
+		_, err = w.Write(expectedInput)
+		assert.NoError(t, err)
+		w.Close()
+		stdin := os.Stdin
+		// Restore stdin right after the test.
+		defer func() { os.Stdin = stdin }()
+		os.Stdin = r
+
+		mockClient.On("ListDeployments", mock.Anything).Return(deploymentRespDefaultQueue, nil).Twice()
+		mockClient.On("GetWorkerQueueOptions").Return(mockWorkerQueueDefaultOptions, nil).Once()
+		mockClient.On("UpdateDeployment", mock.Anything).Return(deploymentRespDefaultQueue[0], nil).Once()
+		cmdArgs := []string{"worker-queue", "create", "-d", "test-deployment-id", "-t", "test-instance-type"}
+		actualOut, err := execDeploymentCmd(cmdArgs...)
+		assert.NoError(t, err)
+		assert.Equal(t, expectedoutput, actualOut)
+		mockClient.AssertExpectations(t)
+	})
+	t.Run("returns an error when getting workspace fails", func(t *testing.T) {
+		testUtil.InitTestConfig(testUtil.Initial)
+		expectedOut := "Usage:\n"
+		cmdArgs := []string{"worker-queue", "create"}
+		resp, err := execDeploymentCmd(cmdArgs...)
+		assert.Error(t, err)
+		assert.NotContains(t, resp, expectedOut)
+	})
+	// TODO When updating existing queues, pass in ID of the existing queues
+	// TODO When updating existing queues, is changing the name of the existing queue/s allowed?
+	// TODO any more error cases
```

Comment:
are these still relevant?

Reply:
Yes leaving them here as they are relevant to #670 which is coming next!